packages/volto/src/components/manage/Blocks/Video/Body.jsx

```diff
@@ -84,8 +98,21 @@ const Body = ({ data, isEditMode }) => {
     tabIndex: 0,
     onKeyPress: onKeyDown,
     ref: ref,
+    id: videoID,
+    source: videoSource,
+    url: videoUrl,
   };
 
+  if (
+    data.url &&
+    peertubeInstances instanceof Array &&
+    data.url.match(new RegExp(peertubeInstances.join('|'), 'gi'))
+  ) {
+    const instance = data.url.match(/^(.*)\/w\/(.*)$/)[1];
+    const videoID = data.url.match(/^(.*)\/w\/(.*)$/)[2];
```

Comment:
- Here, the URL must also start with the domain and have `http`.
- Test a match before accessing the index to avoid `null[0]`

```javascript
  const peertubeRegex = /^https?:\/\/([^/]+)\/w\/([A-Za-z0-9_-]+)/i;
  const match = data.url.match(peertubeRegex);

  if (match) {
    const instance = match[1];
    const videoID = match[2];
```

Reply:
done